templates/contribute/style.md

```diff
@@ -565,3 +565,38 @@ Documentation strings for declarations are delimited with `/-- -/`.
 
 See our [documentation requirements](doc.html) for more suggestions
 and examples.
+
+### Deprecation
+
+Deleting, renaming, or changing declarations can cause downstreams projects that rely on these
+definitions to fail to compile.
+Any publicly exposed theorems and definitions that are being removed should be gracefully
+transitioned by keeping the old declaration with a `@[deprecated]` attribute.
+This warns downstream projects about the change and gives them the opportunity to adjust before the
+declarations are deleted.
+Renamed definitions should use a deprecated `alias` to the new name.
+Otherwise, when the deprecated definition does not have a direct replacement, the definition should
+be deprecated with a message, like so:
+
+```lean4
+theorem new_name : ... := ...
+@[deprecated (since := "YYYY-MM-DD")] alias old_name := new_name
+
+@[deprecated "This theorem is deprecated in favor of using ... with ..." (since := "YYYY-MM-DD")]
+theorem example_thm ...
+```
+
+The `@[deprecated]` attribute requires the deprecation date, and an alias to the new declaration
+or a string to explain how transition away from the old definition when a new version is no longer
+being provided.
+
+The [`deprecate to`](/mathlib4_docs/Mathlib/Tactic/DeprecateTo.html) command and
+`scripts/add_deprecations.sh` script can help generate alias definitions.
+
+Deprecations for declarations with the `to_additive` attribute should ensure the deprecation is
+also tagged with `to_additive`, while for declarations generated by `simps`, this is not strictly
+required.
```

Comment:
I think the `to_additive` one needs an example, and I'm not sure I believe the claim about `simps`.

Reply:
Agreed, an example for to_additive could help.

Reply:
> The sentence about simps goes back to [this message]([#mathlib4 > Deprecation policy @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Deprecation.20policy/near/436744276)) by @fpvandoorn:
> 
> > I would be fine not mandating deprecations for lemmas generated by simps:
> 
> >  - They are generally not typed manually (though they could appear in simp only calls
> 
> >  - It's very hard to have a lemma generated by simps in your code without the original definition, and the simps lemma should change very predictably. (assuming the only change is renaming the def.)
> 
> 

#23959 deprecated a lot of `to_additive` lemmas, and as discussed in the comments, the process was quite a bit more involved than just adding a tag.

Reply:
@apnelson1 I was not aware of this, thanks for the pointer to that PR! I've removed the bit about `simps` and updated with an appropriate example.